### DISCUSSION TITLE
How do I determine compartment number?

### DISCUSSION THREAD

**Discussion by billdenney** [nlmixr2/rxode2#240] [Q&A] [answered]
While working on some stuff for babelmixr2, I want to know how to map from a compartment name to a compartment number.  I found this vignette (https://nlmixr2.github.io/rxode2/articles/rxode2-cmt.html#changing-compartment-numbers-by-pre-declaring-the-compartments), and it suggests that there is a `$state` attribute.  When I looked at an `rxUi` object, I see that `ui$mvFromExpression$state` is there with compartment names.

Is `ui$mvFromExpression$state` the best way to get a name-to-number mapping, or is there a better way?

**Comment by billdenney** [comment, marked as answer]
I think that I found my own answer:  `rxode2::rxModelVars(ui)$state`

**Reply by mattfidler** [reply]
I think it may work on functions too, but am unsure.
